crates/biome_js_analyze/src/lint/nursery/no_misused_promises.rs

```diff
@@ -166,3 +208,88 @@ impl Rule for NoMisusedPromises {
         ))
     }
 }
+
+fn find_misused_promise_expression(
+    expression: &AnyJsExpression,
+    ty: &Type,
+) -> Option<NoMisusedPromisesState> {
+    let parent = expression.syntax().parent()?;
+    let state = match parent.kind() {
+        JsSyntaxKind::JS_CONDITIONAL_EXPRESSION => NoMisusedPromisesState::Conditional,
+        JsSyntaxKind::JS_DO_WHILE_STATEMENT => NoMisusedPromisesState::Conditional,
+        JsSyntaxKind::JS_IF_STATEMENT => NoMisusedPromisesState::Conditional,
+        JsSyntaxKind::JS_SPREAD => NoMisusedPromisesState::Spread,
+        JsSyntaxKind::JS_WHILE_STATEMENT => NoMisusedPromisesState::Conditional,
+        _ => return None,
+    };
+
+    // Uncomment the following line for debugging convenience:
+    //let printed = format!("type of {expression:?} = {ty:?}");
+    let is_promise = ty.is_promise_instance();
+    let is_maybe_promise = ty.has_variant(|ty| ty.is_promise_instance());
+    let should_signal =
+        // "maybe" promises could still be considered conditionals, so we
+        // don't signal those for the conditional variants
+        is_promise || (!matches!(state, NoMisusedPromisesState::Conditional) && is_maybe_promise);
+    should_signal.then_some(state)
+}
+
+fn find_misused_promise_returning_callback(
+    ctx: &RuleContext<NoMisusedPromises>,
+    expression: &AnyJsExpression,
+    ty: &Type,
+) -> Option<NoMisusedPromisesState> {
+    let callback = ty.as_function()?;
+
+    let return_ty = callback.return_type.as_type()?;
+    let return_ty = ty.resolve(return_ty)?;
+
+    let should_signal =
+        return_ty.is_promise_instance() || return_ty.has_variant(|ty| ty.is_promise_instance());
+    if !should_signal {
+        return None;
+    }
+
+    let argument = expression
+        .syntax()
+        .ancestors()
```

Comment:
You can add `.skip(1)` to skip itself

Reply:
This was actually the one case where adding `.skip(1)` would be incorrect, but I did for the others :)